src/Cache/MongoLock.php

```diff
@@ -74,11 +74,11 @@ public function acquire(): bool
             ],
         );
 
-        if (random_int(1, $this->lottery[1]) <= $this->lottery[0]) {
-            $this->collection->deleteMany(['expiration' => ['$lte' => $this->currentTime()]]);
+        if (! empty($this->lottery[0]) && random_int(1, $this->lottery[1]) <= $this->lottery[0]) {
+            $this->collection->deleteMany(['expiration' => ['$lte' => $this->getUTCDateTime()]]);
         }
 
-        return $result['owner'] === $this->owner;
+        return $result->getModifiedCount() > 0 || $result->getUpsertedCount() > 0;
```

Comment:
Wasn't this logic problematic in the original PR? Are you now relying on the UTCDateTime always changing due to sub-second precision?

Reply:
No problem with the previous logic. I said to myself that it was impossible to have the same lock with the same owner acquired in the same microsecond. But I've reverted to that, so you don't have to worry.